external/avro/src/main/scala/org/apache/spark/sql/avro/AvroSerializer.scala

```diff
@@ -165,16 +183,100 @@ class AvroSerializer(rootCatalystType: DataType, rootAvroType: Schema, nullable:
       result
   }
 
-  private def resolveNullableType(avroType: Schema, nullable: Boolean): Schema = {
-    if (nullable) {
+  private def resolveNullableType(avroType: Schema, catalystType: DataType,
```

Comment:
Not caused by this PR, but we better explain what this method does. Can you add a comment for it?